src/components/manager/projects/ProjectJsonSchemaForm.js

```diff
@@ -56,101 +54,83 @@ const ProjectJsonSchemaForm = ({ style, schemaTypes, initialValues, setFileConte
-                        {getSchemaTypeOptions(schemaTypes).map((option) => (
-                            <Select.Option key={option.key} value={option.value}>
-                                {option.text}
-                            </Select.Option>
-                        ))}
-                    </Select>,
-                )}
+        <div>
+            <div {...getRootProps()} style={{
+                border: "2px dashed #bae7ff",
+                textAlign: "center",
+                cursor: "pointer",
+            }}>
+                <input {...getInputProps()} />
+                <p>Drag and drop a JSON Schema file here, or click to select files</p>
+            </div>
+            {value && (
+                <>
+                    <ReactJson src={value || {}} name={false} collapsed={true} />
+                    <Button key="cancel" onClick={() => onChange(null)}>Remove</Button>
+                </>
+            )}
+        </div>
+    );
+};
+JsonSchemaInput.propTypes = {
+    value: PropTypes.object,
+    onChange: PropTypes.func,
+};
+
+const ProjectJsonSchemaForm = ({ form, schemaTypes, initialValues }) => {
+    return (
+        <Form form={form}>
+            <Form.Item
+                label={
+                    <Tooltip title={
+                        <span>The data type on which this <ExtraPropertiesCode tooltip/> schema will be applied</span>
+                    }>
+                        Schema Type
+                    </Tooltip>
+                }
+                name="schemaType"
+                initialValue={initialValues.schemaType}
+                rules={[{ required: true }]}
+            >
+                <Select>
+                    {getSchemaTypeOptions(schemaTypes).map((option) => (
+                        <Select.Option key={option.key} value={option.value}>
+                            {option.text}
+                        </Select.Option>
+                    ))}
+                </Select>
             </Form.Item>
-            <Form.Item label={
-                <Tooltip title={
-                    <span>Check to make the <ExtraPropertiesCode tooltip/> field required</span>
-                }>
-                    <span>Required</span>
-                </Tooltip>
-            }>
-                {form.getFieldDecorator("required", {
-                    initialValue: initialValues.required,
-                    valuePropName: "checked",
-                })(
-                    <Checkbox />,
-                )}
+            <Form.Item
+                label={
+                    <Tooltip title={
+                        <span>Check to make the <ExtraPropertiesCode tooltip/> field required</span>
+                    }>
+                        <span>Required</span>
+                    </Tooltip>
+                }
+                name="required"
+                initialValue={initialValues.required}
+                valuePropName="checked"
+            >
+                <Checkbox />
```

Comment:
Minor: since the `required` field is the only non-mandatory form item, we could place it at the end of the form for consistency, instead of in the middle.